extensions/amp-form/0.1/amp-form.js

```diff
@@ -346,67 +366,100 @@ export class AmpForm {
     return this.isCanonicalAction_ = parseUrl(url).origin == canonicalOrigin;
   }
 
+  handleXhrVerify_() {
+    if (this.state_ === FormState_.SUBMITTING) {
+      return;
+    }
+
+    this.setState_(FormState_.VERIFYING);
+    const verifyXhr = this.doVarSubs_(this.getVarSubsFields_()).then(
+        () => this.doXhr_(map({verify: true})));
```

Comment:
'verify' can collide with existing inputs named `verify`.  Do `__amp_form_verify` and then check if any input has that name same way `SOURCE_ORIGIN_PARAM` is checked in the constructor of AMP form.